### PR TITLE
CTCP-819 Create Location of goods CYA page

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/views/routeDetails/locationOfGoods/CheckYourAnswersView.scala.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    showBackLink = true,
+    lrn = Some(lrn)
+) {
+
+    @heading(
+        heading = messages("routeDetails.locationOfGoods.checkYourAnswers.heading"),
+        caption = Some(messages("routeDetails.caption"))
+    )
+
+    <h2 class="govuk-heading-m">@messages("")</h2>
```
Comment (jackheslop96) [inline app/views/routeDetails/locationOfGoods/CheckYourAnswersView.scala.html:43]: Is this needed?

File: app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        helper.locationOfGoodsUnLocode,
+        helper.additionalIdentifierYesNo,
+        helper.additionalIdentifier,
+        helper.locationOfGoodsIdentification,
+        helper.locationOfGoodsCustomsOfficeIdentifier,
+        helper.locationOfGoodsAddress,
+        helper.locationOfGoodsPostalCode,
+        helper.contactYesNo,
+        helper.contactName,
+        helper.contactPhoneNumber
```
Comment (jackheslop96) [inline app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala:52]: Are these in a suitable order? Should be in the same order as the journey

File: app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import javax.inject.Inject
+
+case class LocationOfGoodsViewModel(section: Section)
+
+object LocationOfGoodsViewModel {
+
+  def apply(userAnswers: UserAnswers)(implicit messages: Messages): LocationOfGoodsViewModel =
+    new LocationOfGoodsViewModelProvider()(userAnswers)
+
+  class LocationOfGoodsViewModelProvider @Inject() () {
```
Comment (jackheslop96) [inline app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala:33]: Is there a spec for this?

File: app/utils/cyaHelpers/routeDetails/LocationOfGoodsCheckYourAnswersHelper.scala [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+import models.reference.{CustomsOffice, UnLocode}
+import models.{Address, Coordinates, LocationOfGoodsIdentification, LocationType, Mode, PostalCodeAddress, UserAnswers}
+import pages.routeDetails.locationOfGoods._
+import pages.routeDetails.locationOfGoods.contact.{LocationOfGoodsContactNamePage, TelephoneNumberPage}
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.Aliases.SummaryListRow
+import utils.cyaHelpers.AnswersHelper
+
+class LocationOfGoodsCheckYourAnswersHelper(userAnswers: UserAnswers, mode: Mode)(implicit messages: Messages) extends AnswersHelper(userAnswers, mode) {
```
Comment (jackheslop96) [inline app/utils/cyaHelpers/routeDetails/LocationOfGoodsCheckYourAnswersHelper.scala:27]: Is there a spec for this?

**Comment by jackheslop96** [comment]
You should also now be able to un-ignore the `NormalMode` tests in `LocationOfGoodsNavigatorSpec`

File: app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala (hunk 3 above)
Comment (kenneth-rayner) [reply app/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModel.scala:33]: There is now

File: test/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModelSpec.scala [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+import models.{Address, Coordinates, LocationOfGoodsIdentification, LocationType, PostalCodeAddress}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import pages.routeDetails.locationOfGoods._
+
+class LocationOfGoodsViewModelSpec extends SpecBase with ScalaCheckPropertyChecks with Generators with RouteDetailsUserAnswersGenerator {
+  "apply" - {
+    "when 'v' customs office" - {
+      "must return 3 rows" in {
+        val userAnswers = emptyUserAnswers
```
Comment (jackheslop96) [inline test/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModelSpec.scala:31]: Thanks for adding this. I wonder if we could do something cool here to utilise the arbitrary answers generator. @kenneth-rayner  do you mind if I pull the branch down and try something?

**Review by jackheslop96 — APPROVED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**